### PR TITLE
feat(projections) - Projecting the search results

### PR DESCRIPTION
**Pull request by akintolga**
- Search results will return only the necessary fields for the list view to function
- There will be another query when the preview panel is opened to get the entire story

### CODE REVIEW

**Comment by akintolga** [comment]
Depends on https://github.com/superdesk/superdesk-core/pull/663

**Comment by akintolga** [comment]
@gbbr Thanks for the points in the previous PR. I have updated the comments. For some reason though when I run grunt docs I don't see the details of the three functions documented in search service. I compared to tags service where it has two functions documented and I can see the details. I'll keep looking.

File: scripts/apps/search/services/SearchService.js [hunk 1]
```diff
@@ -524,6 +523,25 @@ export function SearchService($location, gettext, config, session) {
     };
 
     /**
+     * @ngdoc method
+     * @name search#mergeHighlightFields
+     * @public
+     * @description Merges the highlighted fields to the item
+     * @param {Object} item
+     * @return {Object}
```
Comment (gbbr) [inline scripts/apps/search/services/SearchService.js]: I think `@returns` is the correct form. It might be the reason it's not working? Not sure...

**Comment by akintolga** [comment]
@petrjasek I have changed the list of fields for projection to infer them from the `config.list`. There is a list of `CORE_PROJECTED_FIELDS`  that is required regardless. The others will be inferred from `UI_PROJECTED_FIELD_MAPPINGS` . So we don't need a separate config field for the projected fields.

**Pull request merged**